pkg/cloud/vsphere/actuators/actuators.go

```diff
@@ -25,29 +25,24 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	clustererr "sigs.k8s.io/cluster-api/pkg/controller/error"
 
-	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/constants"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/record"
 )
 
 type patchContext interface {
 	fmt.Stringer
 	GetObject() runtime.Object
 	GetLogger() logr.Logger
-	Patch() error
+	Patch()
```

Comment:
Wondering if we can keep the err return value if we just separate the patch update for spec and status into separate methods

Reply:
Hi @andrewsykim,

I considered it, but it really breaks down into how you consider patching. If patching the object fails, you don't want to update the status, because the status may not reflect the object data known to Kubernetes.

And I've decided that anything you do in the patch should be idempotent anyway, and any subsequent patch should reconcile previous failures.

Additionally, if a patch fails due to a conflict, you pretty much **have** to wait until the next entry. You simply don't have the knowledge to take the updated server view of an object or its status and merged it with your data. That's why the conflict occurred in the first place. 

Which all leads back to this -- patching must be reentrant. If an object or status patch failing causes a provider to enter a corrupt state then the provider's design is flawed.